[SEC-31367] Document Ticket Creation Rules for Automation Pipelines (#36851)
DataDog/documentation: content/en/security/automation_pipelines/create_ticket.md

---
title: Ticket Creation Rules
further_reading:
  - link: "/security/automation_pipelines"
    tag: "Documentation"
    text: "Automation Pipelines"
  - link: "/security/ticketing_integrations"
    tag: "Documentation"
    text: "Ticketing Integrations"
  - link: "/incident_response/case_management"
    tag: "Documentation"
    text: "Case Management"
---

Configure ticket creation rules to automatically create tickets in Jira or Case Management when new findings are discovered. This approach tracks security issues in your existing engineering workflows without manual triage, helping teams respond quickly to new threats at scale. For more information about ticketing integrations with security findings, see [Ticketing Integrations][3].

## Create a ticket creation rule

> 1. On the [Automation Pipelines][2] page, click **Add a New Rule** and select **Create Ticket**.

Reviewer: ```suggestion
1. In Datadog, go to **Security** > **Settings** > [Findings Automation][2]. Click **Add a New Rule**, then select **Create Ticket**. The Create a New Rule page opens.
```

> 1. Enter a descriptive name for the rule, for example, **Critical vulnerabilities for engineering team**.

Reviewer: ```suggestion
1. Under **Rule name**, enter a descriptive name for the rule; for example, "Critical vulnerabilities for engineering team".
```

1. Use the following boxes to configure the rule criteria:
    - **Any of these types**: The types of findings that the rule should check for. Available types include:
      - **Runtime Code Vulnerability**
      - **Static Code Vulnerability**
      - **Library Vulnerability**
      - **Secret**
      - **Infrastructure as Code**
      - **Container Image Vulnerability**
      - **Host Vulnerability**
      - **Misconfiguration**
      - **Attack Path**
      - **Identity Risk**

>       - **API Security**

Reviewer: ```suggestion
      - Runtime Code Vulnerability
      - Static Code Vulnerability
      - Library Vulnerability
      - Secret
      - Infrastructure as Code
      - Container Image Vulnerability
      - Host Vulnerability
      - Misconfiguration
      - Attack Path
      - Identity Risk
      - API Security
```
I think having every bullet point here in bold made it a little hard to read. A small formatting liberty here should help with that 🙂

## Daily ticket limit

Each rule has a configurable daily ticket limit that resets at midnight UTC. When the limit is reached, Datadog creates one final ticket in the same project explaining that the rule hit its daily limit, then stops creating tickets for the remainder of that day. Findings that exceed the limit are not retroactively ticketed when the limit resets, but you can create tickets for them manually.

## Broken rules

If a project configuration error prevents ticket creation—for example, if the connected Jira project is no longer valid—Datadog automatically disables the rule and marks it as broken.

{{< img src="security/automation_pipelines/ticket_creation_broken_rule.png" alt="Automation Pipelines list showing a ticket creation rule with a red warning triangle on a rule, and a tooltip reading 'Rule auto-disabled due to a ticketing integration error'" style="width:100%;" >}}

To resume automatic ticket creation, fix the project configuration and re-enable the rule.

> ## Disabling or deleting a rule

Reviewer: ```suggestion
## Disabled or deleted rules
```
I think this change could help users not think that they're about to see instructions on how to disdable/delete a rule

---
title: Ticket Creation Rules
site_support_id: case_management
further_reading:
  - link: "/security/automation_pipelines"
    tag: "Documentation"
    text: "Automation Pipelines"
  - link: "/security/ticketing_integrations"
    tag: "Documentation"
    text: "Ticketing Integrations"
  - link: "/incident_response/case_management"
    tag: "Documentation"
    text: "Case Management"
---

> {{< product-availability >}}

Reviewer: The [product availability](https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/3532129075/Product+identifiers) shortcode requires products to be defined in the frontmatter to render properly. Let me know if you need help with this!

Configure ticket creation rules to automatically create tickets in Jira or Case Management when new findings are discovered. This approach tracks security issues in your existing engineering workflows without manual triage, helping teams respond quickly to new threats at scale. For more information about ticketing integrations with security findings, see [Ticketing Integrations][3].

## Create a ticket creation rule

1. On the [Automation Pipelines][2] page, click **Add a New Rule** and select **Create Ticket**.
1. Enter a descriptive name for the rule, for example, **Critical vulnerabilities for engineering team**.

> 1. Use the following boxes to configure the rule criteria:

Reviewer: ```suggestion
1. Add your rule criteria into the following fields:
```

    - **Any of these types**: The types of findings that the rule should check for. Available types include:
      - **Runtime Code Vulnerability**
      - **Static Code Vulnerability**
      - **Library Vulnerability**
      - **Secret**
      - **Infrastructure as Code**
      - **Container Image Vulnerability**
      - **Host Vulnerability**
      - **Misconfiguration**
      - **Attack Path**
      - **Identity Risk**
      - **API Security**
    - **Any of these tags or attributes**: The resource tags or attributes that must match for the rule to apply.
1. To add severity criteria to the rule, click **Add Severity**.
1. Select the ticketing system and configure the ticket destination:

    **Jira**
    - **Jira Account**: Select the Atlassian instance to use.
    - **Space**: Select the Jira project. Verify that this space is added to the [Jira Webhook][5].
    - **Ticket Type**: Select the type of Jira issue to create, for example, **Task**.
    - **Assignee** (optional): Specify a user to assign automatically created tickets to.

>     - To add more fields to the Jira ticket Datadog creates, use **Add Optional Field**.

Reviewer: Looks like there are some additional data sync settings that weren't documented?

Author: right, I added them!

    **Case Management**
    - **Case Management Project**: Select an existing Case Management project, or create a new one.

>     - **Assignee** (optional): Specify a user to assign automatically created cases to.

Reviewer: ```suggestion
   - **Jira**
     - **Jira Account**: Select the Atlassian instance to use.
     - **Space**: Select the Jira project. Verify that this space is added to the [Jira Webhook][5].
     - **Ticket Type**: Select the type of Jira issue to create, for example, **Task**.
     - **Assignee** (optional): Specify a user to assign automatically created tickets to.
     - To add more fields to the Jira ticket Datadog creates, use **Add Optional Field**.
   - **Case Management**
     - **Case Management Project**: Select an existing Case Management project, or create one.
     - **Assignee** (optional): Specify a user to assign automatically created cases to.
```
I think these make sense as parallel sub-bullets, so it's easier to visually distinguish them as options.

> 1. Set the **Daily ticket limit**: The maximum number of tickets this rule can create per UTC day.

Reviewer: ```suggestion
1. Under **Rate limit**, enter the [maximum number of tickets](#daily-ticket-limit) this rule can create per UTC day.
```

> 1. Click **Save**. The rule applies to new findings immediately, and ticket creation can take a few minutes after a finding is detected.

Reviewer 2: nit: IMO this sentence is a bit confusing. It says that the rule applies immediately, but it also says that the ticket creation takes a few minutes. That means it's not immediate.

I think it's a simple fix. We can simply say something like "The rule applies to new findings only. It can take up to a few minutes after the finding is detected to create the corresponding ticket."

> 1. Click **Save**. The rule applies to new findings immediately, and ticket creation can take a few minutes after a finding is detected.

Reviewer: ```suggestion
1. Click **Save**. The rule applies to new findings immediately, and tickets become visible a few minutes after Datadog detects a finding.
```
I'm assuming this is kind of meant to be a limitation? I was trying to figure out if it was supposed to be a setup limitation (i.e., your _first_ ticket takes a few minutes to appear), or if it's an ongoing thing users should know about (i.e., _every_ ticket takes a few minutes to appear).

Author: Yes, it's a limitation which occurs for every ticket. I rephrased it (the "become visible" is not exactly accurate, it's really about the ticket creation itself) and also included Aleks' suggestions from the comment above:
> The rule applies to new findings only. It can take up to a few minutes after a finding is detected to create the corresponding ticket.

> **Note**: Ticket creation rules only create tickets for new findings. Existing findings are not retroactively ticketed when you create a rule.

Reviewer: ```suggestion
**Note**: Ticket creation rules only create tickets for new findings. Datadog does not create retroactive tickets for existing findings when you create a rule.
```

## Test a ticket creation rule

After configuring a rule, you can test it to verify the configuration before saving:

1. After filling in the rule form, click **Test Rule**.
1. Select a matching finding to use for the test.
1. Click **Run Test**.

> 1. After the test completes, you can view the created ticket, or detach the test ticket from the finding.

Reviewer: I would add this as a sub-step in the procedure to create a ticket creation rule. Otherwise, the placement on the page suggests that you configure your rule, click save, and then go back and test it. Easy fix!

## Identify automatically created tickets

> {{< img src="security/automation_pipelines/ticket_creation_lightning_indicator.png" alt="Case Management ticket popup showing case VV-3722 was created by an Automation Rule with a lightning bolt indicator, and a link to view all findings with tickets from the same rule" style="width:60%;" >}}

Reviewer: ```suggestion
{{< img src="security/automation_pipelines/ticket_creation_lightning_indicator.png" alt="Case Management ticket popup showing a case created by an Automation Rule, indicated with a lightning bolt icon, and a link to view all findings with tickets that were created from the same rule" style="width:60%;" >}}
```

Tickets created by a rule are marked with a lightning bolt indicator in the findings side panel and explorer views. Hovering over the indicator shows the automation rule responsible for the ticket and provides a link to the rule.

> ## Rule matching order

Reviewer: If you wanted to, I think you could put these sections as subsections in the intro. I think it's helpful for users when they're deciding if they want to use a feature to get the nuance of how it works, and it helps if they can expect conceptual information to appear in one place in the doc. It's not terrible to have it at the end, but it does feel a little more tacked on than the "this is what this thing does!" expectation users already have about overviews.

Author: Thanks for the suggestion, I tried restructuring with those subsections in the intro, but I'm not fully convinced it works well. Those are fairly fine-grained implementation details that most users won't need to read before getting started. The other automation pipeline pages also follow the pattern of putting procedural steps first, which keeps things consistent. I've also kept a link from the rate limit step in the creation flow to the dedicated section, so users who need the detail can jump to it. Happy to revisit if you feel strongly, but I'd lean toward keeping the current structure.

When Datadog identifies a finding, it evaluates the finding against your sequence of ticket creation rules. Starting with the first rule, if there's a match, Datadog creates a ticket using that rule's configuration and stops evaluating further. If no match occurs, Datadog moves to the next rule. This process continues until a match is found or all rules are checked without a match.

## Daily ticket limit

Each rule has a configurable daily ticket limit that resets at midnight UTC. When the limit is reached, Datadog creates one final ticket in the same project explaining that the rule hit its daily limit, then stops creating tickets for the remainder of that day. Findings that exceed the limit are not retroactively ticketed when the limit resets, but you can create tickets for them manually.

## Broken rules

If a project configuration error prevents ticket creation—for example, if the connected Jira project is no longer valid—Datadog automatically disables the rule and marks it as broken.

> {{< img src="security/automation_pipelines/ticket_creation_broken_rule.png" alt="Automation Pipelines list showing a ticket creation rule with a red warning triangle on a rule, and a tooltip reading 'Rule auto-disabled due to a ticketing integration error'" style="width:100%;" >}}

Reviewer: ```suggestion
{{< img src="security/automation_pipelines/ticket_creation_broken_rule.png" alt="Automation Pipelines list showing a ticket creation rule with a warning tooltip that says 'Rule auto-disabled due to a ticketing integration error'" style="width:100%;" >}}
```
